File: graphql-todo-lists-java/src/main/java/com/example/subgraphtodolists/lists/ListsMutations.java
package com.example.subgraphtodolists.lists;

import com.netflix.graphql.dgs.DgsComponent;
import com.netflix.graphql.dgs.DgsMutation;
import org.openapitools.client.models.ToDoList;

import java.io.IOException;

@DgsComponent
public class ListsMutations {
    private final ListsService listsService;

    public ListsMutations(ListsService listsService) {
        this.listsService = listsService;
    }

    @DgsMutation
    public String RemoveList(String listId) throws IOException {
        listsService.removeList(listId);
        return "List removed successfully";
    }

    @DgsMutation
    public ToDoList AddList(String listName) throws IOException {
        return listsService.addList(listName);
    }

    @DgsMutation
    public ToDoList ChangeListName(String listId, String newName) throws IOException {
        return listsService.changeListName(listId, newName);
    }
}

File: graphql-todo-lists-java/src/main/java/com/example/subgraphtodolists/lists/ListsMutations.kt
package com.example.subgraphtodolists.lists

import com.netflix.graphql.dgs.DgsComponent
import com.netflix.graphql.dgs.DgsMutation
import org.openapitools.client.apis.TodoListControllerApi
import org.openapitools.client.models.ListResource
import org.openapitools.client.models.RenameResource
import org.openapitools.client.models.ToDoList
import java.util.UUID

@DgsComponent
class ListsMutations {
    private val api = TodoListControllerApi()

    @DgsMutation
    fun RemoveList(listId: String): String {
        api.delete(UUID.fromString(listId))
        return "List removed successfullyy"
    }

    @DgsMutation
    fun AddList(listName: String): ToDoList {
        val listResource = ListResource(listName)
        return api.createList(listResource)
    }

    @DgsMutation
    fun ChangeListName(listId: String, newName: String): ToDoList {
        val renameResource = RenameResource(newName, UUID.fromString(listId))
        return api.rename(renameResource)
    }
}